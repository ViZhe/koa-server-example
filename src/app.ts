
import Koa from 'koa';
import koaBodyparser from 'koa-bodyparser';
import koaHelmet from 'koa-helmet';
import koaJwt from 'koa-jwt';
import koaLogger from 'koa-logger';
import mongoose from 'mongoose';

import { IS_PRODUCTION, JWT, MONGO } from './config';
import { graphql as gqlServer, rest } from './routes';


mongoose.Promise = Promise;
mongoose.connect(MONGO.MAIN, {
  reconnectInterval: 10000,
  reconnectTries: 1000,
  useCreateIndex: true,
  useNewUrlParser: true,
});
mongoose.set('debug', IS_PRODUCTION);

const app = new Koa();
app.use(koaHelmet());
app.use(koaBodyparser());
app.use(koaLogger());
app.use(rest.routes());
app.use(rest.allowedMethods());
app.use(koaJwt({
  passthrough: true,
  secret: JWT.SECRET,
}));
gqlServer.applyMiddleware({ app });


export default app;
