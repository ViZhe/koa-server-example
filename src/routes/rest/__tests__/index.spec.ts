
import request from 'supertest';

import app from '../../../app';


describe('REST API', () => {
  describe('Path /api', () => {
    let response:request.Response;

    beforeAll(async () => {
      response = await request(app.callback()).get('/api');
    });

    it('should get answer with a code 200', () => {
      expect(response.status).toEqual(200);
    });

    it('should get answer a type json ', () => {
      expect(response.type).toEqual('application/json');
    });

    it('should match body with snapshot', () => {
      expect(response.body).toMatchSnapshot();
    });
  });
});
