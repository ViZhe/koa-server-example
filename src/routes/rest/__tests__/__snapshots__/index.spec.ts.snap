// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`REST API Path /api should match body with snapshot 1`] = `
Object {
  "message": "API method not allowed",
  "status": 405,
  "success": false,
}
`;
